### PR TITLE
[DONT MERGE]initial module view PR

### PR DESCRIPTION
**Pull request by jonathan34c**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/mono/debugger-libs?pullRequest=362) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/mono/debugger-libs?pullRequest=362) before we can accept your contribution.<br/><br/>:x: jonathan34c [sign now](https://cla.dotnetfoundation.org/mono/debugger-libs?pullRequest=362)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/mono/debugger-libs?pullRequest=362) it.</sub>

File: Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs [hunk 1]
```diff
@@ -1304,8 +1308,14 @@ internal protected void OnAssemblyLoaded (string assemblyLocation)
 		{
 			AssemblyLoaded?.Invoke (this, new AssemblyEventArgs (assemblyLocation));
 		}
-		
-		internal protected void SetBusyState (BusyStateEventArgs args)
+
+		internal protected void OnAssemblyAdded (Assembly assembly)
+		{
+			assemblies.Add (assembly);
```
Comment (Therzok) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs:1314]: Assemblies can also be unloaded during a session, so this list needs to support assembly removal as well.

File: Mono.Debugging/Mono.Debugging.Client/AssemblyEventArgs.cs [hunk 2]
```diff
@@ -25,18 +25,24 @@
...
 		{
 			Location = location;
+			Assemblies = assemblies;
 		}
 
 		public string Location {
 			get; private set;
 		}
+
+		public List<Assembly> Assemblies {
```
Comment (Therzok) [inline Mono.Debugging/Mono.Debugging.Client/AssemblyEventArgs.cs:44]: Can we pass the `Assembly` in the event instead? That would also solve the problem of the `assemblies` list having unloaded assemblies in it.

Theoretically, any kind of UI should not need to get all these assemblies in the event arguments, just the latest change.

File: Mono.Debugging.Soft/SoftDebuggerSession.cs [hunk 3]
```diff
@@ -2245,13 +2246,13 @@ void HandleAssemblyLoadEvents (AssemblyLoadEvent[] events)
 			if (events.Length > 1 && events.Any (a => a.Assembly != asm))
 				throw new InvalidOperationException ("Simultaneous AssemblyLoadEvent for multiple assemblies");
 
-			OnAssemblyLoaded(asm.Location);
+			OnAssemblyLoaded(asm.Location, new List<Assembly>());
```
Comment (Therzok) [inline Mono.Debugging.Soft/SoftDebuggerSession.cs:2249]: This should contain an assembly constructed from the event assembly.

File: Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs [hunk 4]
```diff
@@ -49,6 +49,8 @@ public abstract class DebuggerSession: IDisposable
 	{
 		readonly Dictionary<BreakEvent, BreakEventInfo> breakpoints = new Dictionary<BreakEvent, BreakEventInfo> ();
 		readonly Dictionary<string, string> resolvedExpressionCache = new Dictionary<string, string> ();
+		public List<Assembly> assemblies = new List<Assembly> ();
```
Comment (Therzok) [inline Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs:52]: This shouldn't be part of the public API.
```suggestion
	List<Assembly> assemblies = new List<Assembly> ();
```

**Pull request closed without merging**